tests/unit/test_upload.py

```diff
@@ -39,12 +59,25 @@ def test_runtime_error_raised_on_not_being_able_to_connect(self, mock_s3):
         )
 
         with pytest.raises(RuntimeError, match=expected_error):
-            upload.check_aws_access()
+            upload.check_aws_access(slack_alert_webhook="my_webhook")
+
+        with self.subTest("correct Slack message sent"):
+            expected_slack_message = (
+                ":warning:  *S3 Upload*: Error in connecting to AWS!\n\n\t\tAn"
+                " error occurred (1) when calling the bar operation: foo"
+            )
+
+            self.assertEqual(
+                mock_slack.call_args[1]["message"],
+                expected_slack_message,
+            )
 
     @patch("s3_upload.utils.upload.AWS_ACCESS_KEY", "foo")
     @patch("s3_upload.utils.upload.AWS_SECRET_KEY", "bar")
     @patch("s3_upload.utils.upload.AWS_DEFAULT_PROFILE", "baz")
-    def test_system_exit_raised_when_all_env_variables_set(self, mock_s3):
+    def test_system_exit_raised_when_all_env_variables_set(
+        self, mock_s3, mock_slack
+    ):
         expected_error = (
```

Comment:
I didn't add it since both conditionals just set the `error_message` string, then if it is not None it triggers the slack alert to be sent here: https://github.com/eastgenomics/s3_upload/blob/c9b37d780dfa358fa7b69c00b02cbcdab2e83e30/s3_upload/utils/upload.py#L85

I could add an additional subTest for this message so if you would like
<!-- Reviewable comment -OCZGVsm1wX7asQm_Z9F:bm15oef -->
<!-- Sent from Reviewable.io -->

Reply:
Makes sense. They're just different error messages but I see that it just calls the variable so don't mind it not being an additional subTest
<!-- Reviewable comment -OCZIk5oC7xzBQsQkSbj:bbnpusn -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -OCZN9aN8LrjHsehJCZF:b-896fix -->
<!-- Sent from Reviewable.io -->